sdk/keyvault/keyvault/src/secretsModels.ts

```diff
@@ -123,7 +155,7 @@ export interface UpdateSecretOptions extends msRest.RequestOptionsBase {
    * @member {SecretAttributes} [secretAttributes] The secret management
    * attributes.
    */
-  attributes?: SecretAttributes;
+  attributes?: Models.SecretManagementAttributes;

```

Comment:
>attributes?: Models.SecretManagementAttributes; [](start = 2, length = 47)

This should be flattened out. i.e. the writable feilds on SecretAttributes (expires, not before and enabled)  should be fields on the options bag

Reply:
Should I also flatten from `SecretManagementAttributes` into `SecretAttributes`?  This would effectively flatten those three fields in the Secret that users get returned, instead of having an `attribute` field on the secret.

Reply:
Yes we should do this.

Reply:
It should be flattened/unflattened in Secret as well as set/update now.